scripts/template_setup_posix

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env bash
+
+# Automated setup script for Zephyr SDK distribution bundle archive
+# for POSIX-compliant host operating systems (Linux and macOS)
+
+# Utility functions
+pushd()
+{
+  command pushd "$@" > /dev/null
+}
+
+popd()
+{
+  command popd "$@" > /dev/null
+}
+
+# Display script usage
+usage()
+{
+  echo "Usage: $(basename $0) [-c] [-t]"
+  echo
+  echo "  -c     Install Zephyr SDK CMake package"
+  echo "  -t     Install host tools"
+  echo
+  echo "If no arguments are specified, the setup script runs in the interactive"
+  echo "mode asking for user inputs."
+  exit 1
+}
+
+# Ask for user inputs (interactive mode)
+user_prompt()
+{
+  echo "** NOTE **"
+  echo "You only need to run this script once after extracting the Zephyr SDK"
+  echo "distribution bundle archive."
+  echo
+
+  read -n 1 -r -p "Install Zephyr SDK CMake package [y/n]? " yn && echo
+  [ "${yn}" = "y" ] && do_cmake_pkg="y"
+
+  read -n 1 -r -p "Install host tools [y/n]? " yn && echo
+  [ "${yn}" = "y" ] && do_hosttools="y"
+
+  echo
+}
+
+# Entry point
+pushd "$(dirname "${BASH_SOURCE[0]}")"
+
+# Parse arguments
+if [ $# == "0" ]; then
+  interactive="y"
+else
+  while [ "$1" != "" ]; do
+    case $1 in
+      -c)
+        do_cmake_pkg="y"
+        ;;
+      -t)
+        do_hosttools="y"
+        ;;
+      *)
+        echo "Invalid argument '$1'"
+        usage
+        exit 1
+        ;;
+    esac
+    shift
+  done
+fi
+
+# Read bundle version from file
+version=$(<sdk_version)
+
+# Print banner
+echo "Zephyr SDK ${version} Setup"
+echo
+
+# Ask for user inputs if no argument is specified
+if [ "${interactive}" = "y" ]; then
+  interactive="y"
+  user_prompt
+fi
+
+# Install Zephyr SDK CMake package
+if [ "${do_cmake_pkg}" = "y" ]; then
+  echo "Installing Zephyr SDK CMake package ..."
+  cmake -P cmake/zephyr_sdk_export.cmake
```

Comment:
Nothing wrong as such, but we are now expecting CMake to be available when installing.

This means users cannot do:
- install Zephyr SDK
- install CMake

they must now start with CMake.

So should we check that CMake is available and print an warning / error if not ?

Reply:
> they must now start with CMake.

That has always been the case (at least if one is following the Getting Started Guide). In fact, installing a toolchain (i.e. Zephyr SDK) is specified as the last step:
https://docs.zephyrproject.org/latest/getting_started/index.html#install-a-toolchain

> So should we check that CMake is available and print an warning / error if not ?

It will actually print out an error message if CMake is not available now:
```
cmake: command not found
```

Though, a "softer" way of failing would indeed be more desirable. I will look into adding the dependency checks at the top of the scripts.

Reply:
> That has always been the case (at least if one is following the Getting Started Guide). In fact, installing a toolchain (i.e. Zephyr SDK) is specified as the last step:

I know how it's described, but I also know that not everyone reads such guides.
Especially if they bootstrap a new machine with a colleague:
- A: We should install Zephyr-Sdk.
- B: Ok `installing Zephyr-sdk ..... error`
- A: ah, that's right, we must install CMake first.

What we can do is to also add information on how to register Zephyr-SDK in the CMake package registry afterwards.
```
if [ $? -ne 0 ]; then
  echo "ERROR: Registring Zephyr-SDK in CMake package registry failed"
  echo "ERROR: Please run 'cmake -P <path-to>/zephyr_sdk_export.cmake' manually."
fi
```
so that if this fails, they are informed to run this script manually.


In addition, based on the discussion on Discord, should we consider to have the Zephyr-sdk installation check the CMake version.

For example in `zephyr_sdk_export.cmake`:
```
if(CMAKE_VERSION VERSION_LESS 3.20)                                                                                                                                                                                                                                                                                         
  message(WARNING "CMake version: ${CMAKE_VERSION} found.\n"
                  "This version is too old for building Zephyr.\n"
                  "See Zephyr requirements here: https://docs.zephyrproject.org/latest/getting_started/index.html#install-dependencies"
  )                                                                                                                                                                                                                                                                     
endif()  
```

That will give a hint to users if their CMake version is old.
And by referring to the Zephyr docs, the text itself doesn't get outdated so we never print wrong information.
Biggest risk is that if Zephyr bumps min version and we forget to update the check, then there are some versions where the warning is not printed. But when printed, text is always correct.

Those are just minor proposals for improved user experience when things go wrong.
Having enough users, someone will eventually do things in the wrong way so better to be prepared than giving support later :wink: 

Not blocking for this PR in any way.

Reply:
added "dependency check" routine at the beginning of the script, which includes checking the availability of `cmake` in the PATH.

This dependency check routine will eventually be expanded to check for the host tool library dependencies in the future.